packages/fxa-content-server/app/scripts/views/settings.js

```diff
@@ -142,7 +152,21 @@ const View = BaseView.extend({
   beforeRender() {
     const account = this.getSignedInAccount();
 
-    return account.fetchProfile().then(() => this.user.setAccount(account));
+    return Promise.all([
+      account.fetchProfile(),
+      this.user.setAccount(account),
+      account.settingsData(),
+    ])
+      .then(([, , data]) => {
```

Comment:
This looks like it fetches the profile, sets an account, and fetches settings data in parallel. The original line fetched the profile and then set the account in series. Is the change to parallel fetch intentional? (i.e. if fetched profile data is a dependency of `setAccount`, we may end up with a race condition here)

Reply:
They're all independent operations. The decision to run them concurrently comes from https://github.com/mozilla/fxa/pull/1846#discussion_r306221340.

Reply:
If it's a problem, maybe this should be something more like:
```javascript
    return Promise.all([
      account.fetchProfile().then(() => this.user.setAccount(account)),
      account.settingsData(),
    ])
      .then(([, data]) => {
```

Reply:
There's no need for `fetchProfile` and `setAccount` to be run serially, unless I'm missing something? They're completely independent.

Reply:
Yeah, I'm fuzzy on whether there's a real problem here (I didn't see one)